### PR TITLE
SALTO-1904: add custom fields references in zendesk

### PR DESCRIPTION
**Pull request by noamhammer**
_Add custom fields references in zendesk_

---

_Additional context for reviewer_
None

---
_Release Notes_: 
_Zendesk-Support Adapter_
* Add references to custom ticket fields in views and macros

---
_User Notifications_: 
_Add references to custom ticket fields in views and macros_

### CODE REVIEW

File: packages/zendesk-support-adapter/src/filters/field_references.ts [hunk 1]
```diff
@@ -52,6 +52,26 @@ const getLowerCaseSingularLookupType = (val: string): string | undefined => {
...
+type ZendeskSupportReferenceSerializationStrategyName = 'customFields'
+const ZendeskSupportReferenceSerializationStrategyLookup: Record<
+  ZendeskSupportReferenceSerializationStrategyName
+  | referenceUtils.ReferenceSerializationStrategyName,
+  referenceUtils.ReferenceSerializationStrategy
+> = {
+  ...referenceUtils.ReferenceSerializationStrategyLookup,
+  customFields: {
+    serialize: ({ ref }) => (isInstanceElement(ref.value)
+      ? CUSTOM_FIELDS_PREFIX.concat(ref.value.value.id?.toString())
```
Comment (netama) [inline packages/zendesk-support-adapter/src/filters/field_references.ts:66]: nit: maybe 
```
`${CUSTOM_FIELDS_PREFIX}${...}`
```

File: packages/zendesk-support-adapter/src/filters/field_references.ts [hunk 2]
```diff
@@ -63,9 +83,25 @@ export const contextStrategyLookup: Record<
   parentValue: neighborContextFunc({ contextFieldName: 'value', levelsUp: 2, contextValueMapper: getValueLookupType }),
 }
 
-export const fieldNameToTypeMappingDefs: referenceUtils.FieldReferenceDefinition<
-  ReferenceContextStrategyName
->[] = [
+type ZendeskSupportFieldReferenceDefinition = referenceUtils
+  .FieldReferenceDefinition<ReferenceContextStrategyName> & {
```
Comment (netama) [inline packages/zendesk-support-adapter/src/filters/field_references.ts:87]: nit: can we avoid the line break before the `.` (and do it inside the brackets or something)? (and same for below)

File: packages/adapter-components/src/references/field_references.ts [hunk 3]
```diff
@@ -228,8 +228,12 @@ export const addReferences = async <
...
   T extends string,
   GenericFieldReferenceDefinition extends FieldReferenceDefinition<T>
->(defs: GenericFieldReferenceDefinition[]): GetLookupNameFunc => {
-  const resolverFinder = generateReferenceResolverFinder(defs)
+>(
+    defs: GenericFieldReferenceDefinition[],
+    fieldReferenceResolverCreator?: (def: GenericFieldReferenceDefinition) =>
+      FieldReferenceResolver<T>,
+  ): GetLookupNameFunc => {
+  const resolverFinder = generateReferenceResolverFinder(defs, fieldReferenceResolverCreator)
```
Comment (netama) [inline packages/adapter-components/src/references/field_references.ts:236]: do we need this change if you're still using the additional arg like in zuora?

**Review by netama — APPROVED** [review]
LGTM but I think the adapter-components change is not needed? (if it is, please add a unittest for it with an example)

File: packages/adapter-components/src/references/field_references.ts (hunk 3 above)
Comment (noamhammer) [reply packages/adapter-components/src/references/field_references.ts:236]: for the deploy

**Pull request merged**